perun-registrar-lib/src/main/java/cz/metacentrum/perun/registrar/impl/ConsolidatorManagerImpl.java

```diff
@@ -600,7 +777,19 @@ else if (ues.getExtSource().getName().equals("https://login.elixir-czech.org/idp
 						} catch (AttributeNotExistsException | WrongAttributeAssignmentException ex) {
 							// don't care
 						}
-						es.add(ues.getExtSource());
+						try {
+							nameAttr = perun.getAttributesManagerBl().getAttribute(registrarSession, ues, AttributesManager.NS_UES_ATTR_DEF + ":sourceIdPName");
+						} catch (Exception e) {
+							log.debug("Failed to get sourceIdpName attribute" + e);
+						}
+						var richExtSource = new RichExtSource(ues.getExtSource());
+						if (nameAttr != null) {
+							richExtSource.setAttributes(List.of(nameAttr));
+						} else {
+							richExtSource.setAttributes(Collections.emptyList());
+						}
+						es.add(richExtSource);
```

Comment:
Adding sourceIdPName attribute to RichIdentity - this method is called by the `checkForSimilarUsers` which I am calling from consolidator to get similar users